static_src/stores/app_store.js

```diff
@@ -107,8 +110,19 @@ class AppStore extends BaseStore {
         break;
       }
 
+      case appActionTypes.APP_START: {
+        const app = this.get(action.appGuid);
+        if (app) {
+          const startingApp = Object.assign({}, app,
+            { state: appStates.starting });
+          this.merge('guid', startingApp, (changed) => {
+            if (changed) this.emitChange();
+          });
```

Comment:
The callback shouldn't be necessary, `merge` uses the [defaultChangedCallback](https://github.com/18F/cg-dashboard/blob/master/static_src/stores/base_store.js#L13) which handles this for you.